### ISSUE TITLE
data preprocessing

### ISSUE THREAD

**Issue by Wang-5ying** [yejees/APT#1]
Could you please confirm whether the following data processing pipeline was adopted prior to model training: raw BraTS data → nnUNet preprocessing → final dataset?

**Comment by yejees** [comment]
Thank you for your attention. We do not use the nnUNet preprocessing; instead, we apply a simple transformation to shift the data distribution, data * 2 - 1.

**Comment by Wang-5ying** [comment]
Thank you for your response. I have a further question regarding the testing procedure.
In the test code, I did not observe a direct invocation of the prior codebook proposed in Section 3.3 of the paper. Instead, an initial value appears to be manually inferred during the testing stage.

Could you please clarify whether this implementation is equivalent to utilizing the prior codebook, or if the effect of the prior codebook has been implicitly incorporated into the initialization process?

Also, I noticed this line in the code:
optimizer = optim.Adam(model.parameters(), lr=5e-5, betas=(0.9,0.98), eps=1e-6, weight_decay=0.2)
Is the weight_decay value of 0.2 intentionally large?

**Comment by yejees** [comment]
Thank you for your detailed questions. 

Regarding the prior codebook:
In the current test script, the effect of the prior codebook is implicitly incorporated in the inference code, rather than being invoked as a separate explicit module. The provided implementation is tailored to the BraTS dataset, for which the prior codebook has already been reflected in the initialization used at test time.

If you would like to apply the prior codebook mechanism to a different dataset, you can run the prior_codebook.py script yourself and then adapt the generated prior image to define the corresponding initialization in the inference pipeline.

As for the optimizer setting:
The weight_decay value of 0.2 is not the result of a careful, dataset-specific tuning process on our side. It was initially adopted by following another reference implementation. You should feel free to adjust and retune weight_decay for your own experiments, and a smaller value may well be more appropriate depending on the setup.

**Comment by Wang-5ying** [comment]
Thank you for your previous responses. I have a follow-up question regarding the dataset and reproducibility.

In your paper, it is mentioned that the BraTS 2021 dataset comprises 2,040 cases, with 80% used for training and 50 subjects (200 slices) for testing. However, the official BraTS 2021 repository only provides 1,251 cases for training and 291 for validation/testing.

Due to this discrepancy, my reproduction using the official split fails to match the PSNR results reported in your paper. Could you please provide:

Model Weights: The pre-trained weights for both the Stage-1 and Stage-2 models.

Test Set Details: The list of the 50 specific subjects (or the processed 200 slices) used for your evaluation.

Thank you again for your time and assistance in helping me reproduce your work.
